shared/src/sector/seal.rs

```diff
@@ -33,6 +33,18 @@ pub struct SealVerifyInfo {
     pub unsealed_cid: Cid, // Commd
 }
 
+/// Information needed to verify a non-interactive seal proof.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize_tuple, Deserialize_tuple)]
+pub struct NISealVerifyInfo {
+    pub registered_proof: RegisteredSealProof,
+    pub sector_id: SectorID,
+    pub randomness: SealRandomness,
+    #[serde(with = "strict_bytes")]
+    pub proof: Vec<u8>,
+    pub sealed_cid: Cid,   // Commr
+    pub unsealed_cid: Cid, // Commd
+}
```

Comment:
Can we do this without introducing a new NISealVerifyInfo type, and thus without a new syscall at all? The proof type carries the information that it is NI, and then determines that certain fields will be ignored.
- deal_ids: already ignored and always empty
- interactive_randomness: let's just specify it shall be zero (and ignored) for NI proofs

Reply:
Yeah, that was my initial thinking too. Will change that.